app/src/lib/shells/win32.ts

```diff
@@ -268,6 +284,38 @@ async function findCygwin(): Promise<string | null> {
   return null
 }
 
+async function findWSL(): Promise<string | null> {
+  const system32 = Path.join(
+    process.env.SystemRoot || 'C:\\Windows',
+    'System32'
+  )
+  const wslPath = Path.join(system32, 'wsl.exe')
+  const wslConfigPath = Path.join(system32, 'wslconfig.exe')
+
+  if (!(await pathExists(wslPath))) {
+    log.debug(`[WSL] wsl.exe does not exist at '${wslPath}'`)
+    return null
+  }
+  if (!(await pathExists(wslConfigPath))) {
+    log.debug(
+      `[WSL] found wsl.exe, but wslconfig.exe does not exist at '${wslConfigPath}'`
+    )
+    return null
+  }
+  const exitCode = new Promise<number>((resolve, reject) => {
+    const wslDistros = spawn(wslConfigPath, ['/list'])
+    wslDistros.on('error', reject)
+    wslDistros.on('exit', resolve)
+  })
+  if ((await exitCode) !== 0) {
```

Comment:
I know I mentioned making this promise `reject` if the underlying list `spawn` failed, but I think this now means we need to handle the error here.

Can we wrap this in a `try-catch` block and log these two scenarios:

 - the error that gets raised when the `await` throws because of a rejected promise
 - the non-zero exit code returned when the promise resolves

Something like this:

```ts
try {
  const result = await exitCode
  if (result !== 0) {
    log.debug(
      `[WSL] found wsl.exe and wslconfig.exe, but no distros are installed. Error code: ${result}`
    )
    return null
  }

  return wslPath
} catch (err) {
    log.error(`[WSL] unhandled error when invoking 'wsl /list'`, err)
} 
```

Another benefit of this change is we'll fix how the `log.debug` tries to write the promise itself to the log, when we want the result of the promise instead.

Reply:
I added the suggestion above, and I also return null if we reach the end of the method as if we make it out of the try catch block then something went wrong anyway. I can put the return into the catch block if needed, but I figured this would handle if something weird happened.